core/src/main/java/org/apache/calcite/rel/RelNode.java

```diff
@@ -407,6 +407,17 @@ RelNode copy(
    */
   void register(RelOptPlanner planner);
 
+  /**
+   * Indicates whether it is an enforcer operator, e.g. PhysicalSort,
+   * PhysicalHashDistribute, etc. As an enforcer, the operator must be
+   * created only when required traitSet is not satisfied by its input.
+   *
+   * @return Whether it is an enforcer operator
+   */
+  default boolean isEnforcer() {
+    return false;
```

Comment:
Should we add this default interface ? There is already a default value in `AbstractRelNode#isEnforcer`.

Reply:
we can remove AbstractRelNode. isEnforcer